source/Octopus.Tentacle.Tests.Integration/ClientScriptExecutionCanBeCancelledWhenRetriesAreDisabled.cs

```diff
@@ -134,22 +131,26 @@ await syncOrAsyncHalibut.WhenSync(() => scriptServiceV2.EnsureTentacleIsConnecte
         }
 
         [Test]
-        [SyncAndAsyncTestCase(TentacleType.Polling, RpcCallStage.Connecting, ExpectedFlow.CancelRpcAndExitImmediately)]
-        [SyncAndAsyncTestCase(TentacleType.Listening, RpcCallStage.Connecting, ExpectedFlow.CancelRpcAndExitImmediately)]
-        [SyncAndAsyncTestCase(TentacleType.Polling, RpcCallStage.InFlight, ExpectedFlow.AbandonRpcThenCancelScriptThenCompleteScript)]
-        [SyncAndAsyncTestCase(TentacleType.Listening, RpcCallStage.InFlight, ExpectedFlow.AbandonRpcThenCancelScriptThenCompleteScript)]
-        public async Task DuringStartScript_ScriptExecutionCanBeCancelled(TentacleType tentacleType, RpcCallStage rpcCallStage, ExpectedFlow expectedFlow, SyncOrAsyncHalibut syncOrAsyncHalibut)
+        [TentacleConfigurations(additionalParameterTypes: new object[] {typeof(RpcCallStage)})]
```

Comment:
Nit: I like how we can pass in extra source enumerables. But I can see here in this test how we can't use that to pass in an expected flow.

This made me wonder, did we want to consider splitting `additionalParameterTypes` into two separate optional parameters? E.g. something like `additionalSourceTypes` and `additionalParameters`.

Where the first does what it currently does, and the second simply uses the values it is given.  Then this test could look like:

```
[TentacleConfigurations(additionalParameters: new object[] {RpcCallStage.Connecting, ExpectedFlow.CancelRpcAndExitImmediately})]
[TentacleConfigurations(additionalParameters: new object[] {RpcCallStage.InFlight, ExpectedFlow.AbandonRpcThenCancelScriptThenCompleteScript})]
```

This would save having to do the switch below.

Just a thought 🙂

Reply:
If we do this, then we can remove the switch statements from this file, and other files. I won't bother tagging them all.

Reply:
We certainly could do this, but we only use the switch statement in a handful of tests, and I'm not sure it's worth the added complexity and rework to build this into the attribute.

Let's leave it as-is for now, and if we need to do this kind of thing more in the future, we can add it then.